examples/image_reconstruction.ipynb

```diff
@@ -0,0 +1,705 @@
+      "Requirement already satisfied: future in /usr/local/lib/python3.6/dist-packages (from torch->poutyne) (0.16.0)\n",
+      "Requirement already satisfied: typing-extensions in /usr/local/lib/python3.6/dist-packages (from torch->poutyne) (3.7.4.3)\n"
+     ]
+    }
+   ],
+   "source": [
+    "pip install poutyne"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "Uwt_Nl1Qw_Pz"
+   },
+   "outputs": [],
+   "source": [
+    "#importing the needed libraries\n",
+    "\n",
+    "%matplotlib inline\n",
+    "%reload_ext autoreload\n",
+    "%autoreload 2\n",
+    "\n",
+    "import numpy as np\n",
+    "import math\n",
+    "import os\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "\n",
+    "import torch\n",
+    "import torch.nn as nn\n",
+    "import torch.optim as optim\n",
+    "from torch.utils.data import DataLoader,Subset, Dataset\n",
+    "\n",
+    "\n",
+    "import torchvision\n",
+    "import torchvision.datasets as datasets\n",
+    "import torchvision.transforms as tfms\n",
+    "from torchvision.utils import make_grid\n",
+    "\n",
+    "\n",
+    "import poutyne\n",
+    "from poutyne import set_seeds, Model, Experiment, ModelCheckpoint,CSVLogger\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "j7N7G1IWw_P0",
+    "outputId": "3f03c676-b228-44cb-95a3-594b7e3abc9b"
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The running processor is... cuda\n"
+     ]
+    }
+   ],
+   "source": [
+    "# hyper parameters\n",
+    "\n",
+    "num_epochs = 5\n",
+    "learning_rate = 0.001\n",
+    "batch_size = 32\n",
+    "image_size = 224\n",
+    "val_split_percent = 0.2\n",
+    "momentum = 0.5\n",
+    "device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')\n",
+    "print ('The running processor is...', device)\n",
+    "set_seeds(42)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "M56Tvbbvw_P1"
+   },
+   "source": [
+    "The MNIST dataset is directly donlowded from torchvision.datasets. The training dataset contains 60,000 images of digits at the size of 28x28, However we seperate 20% of the full train dataset as validation dataset. On the other hand by passing \"train\" argument as False, the test dataset, containing 10,000 images is downloaded and saved in the dataset directory."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "QVJ3WG_hw_P1"
+   },
+   "outputs": [],
+   "source": [
+    "# This section produces the original MNIST dataset, which is provided for classification applications. ---> (image, lable)\n",
+    "# In the next section we should redefine our datasets to make it proper for the reconstruction application. ---> (image, image)\n",
```

Comment:
I don't understand the need for a comment if we have a block of text before. Feel like a section people will not read.